### PR TITLE
feat: organise settings by categories [IDE-437]

### PR DESCRIPTION
**Pull request by cat2608**
### Description

This PR reorganises the Snyk settings page into categorised sections, following a similar structure to the IntelliJ Settings page. 

The new organisation aims to enhance user experience by making the settings more intuitive and easier to navigate. The main categories introduced are:

- **General Settings**
- **Product Selection**
- **Severity Selection**
- **Project Settings**
- **Executable Settings**
- **User Experience**
- **Advanced**

### Checklist

- [ ] Tests added and all succeed
- [ ] Linted
- [ ] CHANGELOG.md updated
- [ ] README.md updated, if user-facing

### Screenshots / GIFs

image

https://github.com/user-attachments/assets/5521b21c-d891-42c4-b7bd-766cc7634175

### CODE REVIEW

File: package.json
Comment (bastiandoetsch) [inline package.json:1]: How do the sub sections affect configuration updates to Language Server when it pulls the config via `DidChangeConfiguration`?

File: package.json [hunk 1]
```diff
@@ -53,8 +53,34 @@
...
+          },
+          "snyk.advanced.customEndpoint": {
+            "type": "string",
+            "markdownDescription": "Sets API endpoint to use for Snyk requests. Useful for custom Snyk setups. E.g. `https://api.eu.snyk.io`.",
+            "scope": "window",
+            "pattern": "^(|(https?://)api.*.(snyk|snykgov).io)$"
+          },
+          "snyk.advanced.organization": {
+            "type": "string",
+            "markdownDescription": "Specifies an organization slug name to run tests for that organization. \n\nNote: The slug name can be extracted from the URL of your organization in the Snyk UI: `https://app.snyk.io/org/[orgslugname]`. If not specified, preferred organization as defined in your [web account settings](https://app.snyk.io/account) is used to run tests.",
```
Comment (bastiandoetsch) [inline package.json:81]: UUID would be better as it's unique and always works. Should we update the copy to contain instructions on how to retrieve the UUID?

**Review by bastiandoetsch — COMMENTED** [review]
Very nice! Let's get also feedback from @neilm-snyk or @andygongea

**Comment by bastiandoetsch** [comment]
Also, let's update the changelog, as this change is very user-visible.

File: package.json
Comment (bastiandoetsch) [reply package.json:1]: When testing manually, it seems to work fine :)

File: package.json (hunk 1 above)
Comment (cat2608) [reply package.json:81]: Good idea! 
OMW 💪

Comment (cat2608) [reply package.json]: My copy suggestion:

|Before|After|
|-|-|
|with-org-id|with-slug|

I'm going to double-check it with Neil and Andy:

**Pull request merged**